core/src/test/scala/org/ensime/indexer/FileWatcherSpec.scala

```diff
@@ -140,11 +167,13 @@ abstract class FileWatcherSpec extends EnsimeSpec
         try {
           withClassWatcher(dir) { watcher =>
             // would be better if this was atomic (not possible from JVM?)
-            parent.tree.reverse.foreach(_.delete())
+            waitForBaseRegistered(tk)
 
+            parent.tree.reverse.foreach(_.delete())
```

Comment:
I've seen this code multiple times - utility method?
Originally there was FileUtils.deleteDirectory - maybe something similar?